app/views/folders/index.html.erb

```diff
@@ -1,3 +1,3 @@
 <% @folders.each do |folder| %>
-  <%= link_to folder.name, folder_path(current_user.username, folder.route) %>
+  <%= link_to folder.name, folder_path(folder.owner.username, folder.route) %>
```

Comment:
This path helper works, but there is the `folder.url` method which would make it more concise.